cpp/foxglove-websocket/include/foxglove/websocket/websocket_server.hpp

```diff
@@ -554,31 +565,49 @@ inline void Server<ServerConfiguration>::handleMessage(ConnHandle hdl, MessagePt
   try {
     switch (op) {
       case OpCode::TEXT: {
-        handleTextMessage(hdl, msg->get_payload());
+        _handlerCallbackQueue->addCallback([this, hdl, msg]() {
```

Comment:
seems like we could deduplicate some code if we move the `switch` inside of the callback. What was the reason for changing the signature of the handler functions?